EventFilter/CastorRawToDigi/src/ZdcUnpacker.cc

```diff
@@ -12,236 +13,278 @@
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include <map>
 
-namespace ZdcUnpacker_impl {
+
+namespace HcalUnpacker_impl {
   template <class DigiClass>
-  const HcalQIESample* unpack(const HcalQIESample* startPoint, const HcalQIESample* limit, DigiClass& digi, int presamples, const CastorElectronicsId& eid, int startSample, int endSample, int expectedTime, const HcalHTRData& hhd) {
+  const unsigned short* unpack_compact(const unsigned short* startPoint, const unsigned short* limit, DigiClass& digi, 
+				       int presamples, const HcalElectronicsId& eid, int startSample, int endSample, 
+				       int expectedTime, const HcalHTRData& hhd) {
     // set parameters
     digi.setPresamples(presamples);
-    int fiber=startPoint->fiber();
-    int fiberchan=startPoint->fiberChan();
-    uint32_t zsmask=hhd.zsBunchMask()>>startSample;
-    digi.setZSInfo(hhd.isUnsuppressed(),hhd.wasMarkAndPassZS(fiber,fiberchan),zsmask);
+    digi.setReadoutIds(eid);
+    int flavor, error_flags, capid0, channelid;
 
-    if (expectedTime>=0 && !hhd.isUnsuppressed()) {
-      digi.setFiberIdleOffset(hhd.getFibOrbMsgBCN(fiber)-expectedTime);
-    }
-    int myFiberChan=startPoint->fiberAndChan();
-    int ncurr=0,ntaken=0;
-    const HcalQIESample* qie_work=startPoint;
-    while (qie_work!=limit && qie_work->fiberAndChan()==myFiberChan) {
-      if (ncurr>=startSample && ncurr<=endSample) {
-        digi.setSample(ntaken,*qie_work);
-        ++ntaken;
-      }
-      ncurr++;
-      qie_work++;
-    }
-    digi.setSize(ntaken);
-    return qie_work;
-  }
-}
+    HcalHTRData::unpack_per_channel_header(*startPoint,flavor,error_flags,capid0,channelid);
+    bool isCapRotating=!(error_flags&0x1);
```

Comment:
No, this is also just a copy from HcalUnpacker
https://github.com/cms-sw/cmssw/blob/CMSSW_7_5_X/EventFilter/HcalRawToDigi/src/HcalUnpacker.cc#L56